### ISSUE TITLE
glob should be allowed to return bool

### ISSUE THREAD

**Issue by matbech** [peachpiecompiler/peachpie#944]
I get this warning:
warning PHP5011: Unreachable code detected

with this code here:
```php
$res = glob($pattern, $globFlags);
if ($res === false) {
   throw new Exception\RuntimeException("glob('{$pattern}', {$globFlags}) failed", 0, $err);
}
```

However, glob is expected to return false (bool) on error:
https://www.php.net/manual/en/function.glob.php

I do assume the glob implementation in PeachPie is throwing an exception on error, instead of false.

**Comment by jakubmisek** [comment]
also, the internal implementation just ignores exceptions and does not validate input arguments well
